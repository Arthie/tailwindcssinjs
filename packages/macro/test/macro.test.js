const pluginTester = require("babel-plugin-tester");
const plugin = require("babel-plugin-macros");

pluginTester.default({
  plugin,
  snapshot: true,
  babelOptions: {
    filename: __filename,
  },
  tests: {
    "no usage": `import tw from '../lib/macro'`,
    "correct usage 1": `
      import tw from '../lib/macro';
      const css = tw('w-1/2 bg-blue-500 scale-0 active:bg-red-300 lg:hover[text-red-200 bg-blue-500]', ['lg:text-red-300',"focus:bg-green-300"], "md:bg-green-300");
    `,
    "correct usage 2": `
      import tw from '../lib/macro';
      const css = tw("m-0 mr-4 text-red-400 transform -translate-y-1/2 lg:hover[bg-blue-100 text-red-200]");
    `,
    "correct usage tagged": `
      import tw from '../lib/macro';
      const css = tw\`supports-grid:text-blue-400 important[text-green-200 bg-red-300] m-0 mr-4 text-red-400 transform -translate-y-1/2 lg:hover[bg-blue-100 text-red-200]\`;
    `,
    "tailwind-ui ussage": `
      import tw from '../lib/macro';
      const css = tw\`relative w-full flex justify-center py-2 px-4 border border-transparent text-sm leading-5 font-medium rounded-md text-white bg-indigo-600 hover:bg-indigo-500 focus:outline-none focus:border-indigo-700 focus:shadow-outline-indigo active:bg-indigo-300 transition duration-150 ease-in-out\`;
    `,
    "test screen": `
      import tw from '../lib/macro';
      const css = tw\`md2[text-red-200] bg-red-300\`;
    `,
    "components": `
      import tw from '../lib/macro';
      const css = tw\`container form-input\`
    `,
    "variant class not allowed": {
      code: `
      import tw from '../lib/macro';
      const css = tw\`md[container] form-input\`
      `,
      snapshot: true,
      error: /Variant class/,
    },
    "class not found error": {
      code: `
      import tw from '../lib/macro';
      const css = tw\`lol\`
      `,
      snapshot: true,
      error: /not found/,
    },
    "version 1.3.X test 1": `
      import tw from '../lib/macro';
      const css = tw\`group-focus:text-red-100 delay-1000 space-y-4 flex space-x-2 divide-x divide-y divide-blue-500 space-x-reverse\`
      `,
    "version 1.3.X test 2": `
      import tw from '../lib/macro';
      const css = tw\`group-focus:text-red-100 md:delay-1000 md:space-y-4 md:space-x-2 md:divide-x md:divide-y md:divide-blue-500 md:space-x-reverse\`
      `,
  },
});