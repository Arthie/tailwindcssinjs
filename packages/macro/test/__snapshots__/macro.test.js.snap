// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`macros class not found error: class not found error 1`] = `

import tw from '../lib/macro';
const css = tw\`lol\`

      ↓ ↓ ↓ ↓ ↓ ↓

MacroError: /home/arthie/projects/tailwindcssinjs/packages/macro/test/macro.test.js: Error: Class "lol" not found. Check Tailwind config. Restart dev server and or clearing babel cache after Tailwind config changes.

`;

exports[`macros components: components 1`] = `

import tw from '../lib/macro';
const css = tw\`container form-input\`

      ↓ ↓ ↓ ↓ ↓ ↓

const css = {
  width: "100%",
  appearance: "none",
  backgroundColor: "#ffffff",
  borderColor: "#d2d6dc",
  borderWidth: "1px",
  borderRadius: "0.375rem",
  paddingTop: "0.5rem",
  paddingRight: "0.75rem",
  paddingBottom: "0.5rem",
  paddingLeft: "0.75rem",
  fontSize: "1rem",
  lineHeight: "1.5",
  "&::placeholder": {
    color: "#9fa6b2",
    opacity: "1",
  },
  "&:focus": {
    outline: "none",
    boxShadow: "0 0 0 3px rgba(164, 202, 254, 0.45)",
    borderColor: "#a4cafe",
  },
  "@media (min-width: 668px)": {
    maxWidth: "668px",
  },
  "@media (min-width: 868px)": {
    maxWidth: "868px",
  },
  "@media (min-width: 640px)": {
    maxWidth: "640px",
  },
  "@media (min-width: 1024px)": {
    maxWidth: "1024px",
  },
  "@media (min-width: 1280px)": {
    maxWidth: "1280px",
  },
  "@media (min-width: 1536px)": {
    maxWidth: "1536px",
  },
};


`;

exports[`macros correct usage 1: correct usage 1 1`] = `

import tw from '../lib/macro';
const css = tw('w-1/2 bg-blue-500 scale-0 active:bg-red-300 lg:hover[text-red-200 bg-blue-500]', ['lg:text-red-300',"focus:bg-green-300"], "md:bg-green-300");

      ↓ ↓ ↓ ↓ ↓ ↓

const css = {
  "--transform-scale-x": "0",
  "--transform-scale-y": "0",
  width: "50%",
  backgroundColor: "#3f83f8",
  "&:active": {
    backgroundColor: "#f8b4b4",
  },
  "&:focus": {
    backgroundColor: "#84e1bc",
  },
  "@media (min-width: 1024px)": {
    backgroundColor: "#84e1bc",
  },
  "@media (min-width: 1280px)": {
    color: "#f8b4b4",
    "&:hover": {
      color: "#fbd5d5",
      backgroundColor: "#3f83f8",
    },
  },
};


`;

exports[`macros correct usage 2: correct usage 2 1`] = `

import tw from '../lib/macro';
const css = tw("m-0 mr-4 text-red-400 transform -translate-y-1/2 lg:hover[bg-blue-100 text-red-200]");

      ↓ ↓ ↓ ↓ ↓ ↓

const css = {
  "--transform-translate-x": "0",
  "--transform-translate-y": "-50%",
  "--transform-rotate": "0",
  "--transform-skew-x": "0",
  "--transform-skew-y": "0",
  "--transform-scale-x": "1",
  "--transform-scale-y": "1",
  margin: "0",
  marginRight: "1rem",
  color: "#f98080",
  transform:
    "translateX(var(--transform-translate-x)) translateY(var(--transform-translate-y)) rotate(var(--transform-rotate)) skewX(var(--transform-skew-x)) skewY(var(--transform-skew-y)) scaleX(var(--transform-scale-x)) scaleY(var(--transform-scale-y))",
  "@media (min-width: 1280px)": {
    "&:hover": {
      backgroundColor: "#e1effe",
      color: "#fbd5d5",
    },
  },
};


`;

exports[`macros correct usage tagged: correct usage tagged 1`] = `

import tw from '../lib/macro';
const css = tw\`supports-grid:text-blue-400 important[text-green-200 bg-red-300] m-0 mr-4 text-red-400 transform -translate-y-1/2 lg:hover[bg-blue-100 text-red-200]\`;

      ↓ ↓ ↓ ↓ ↓ ↓

const css = {
  "--transform-translate-x": "0",
  "--transform-translate-y": "-50%",
  "--transform-rotate": "0",
  "--transform-skew-x": "0",
  "--transform-skew-y": "0",
  "--transform-scale-x": "1",
  "--transform-scale-y": "1",
  margin: "0",
  marginRight: "1rem",
  color: "#f98080",
  transform:
    "translateX(var(--transform-translate-x)) translateY(var(--transform-translate-y)) rotate(var(--transform-rotate)) skewX(var(--transform-skew-x)) skewY(var(--transform-skew-y)) scaleX(var(--transform-scale-x)) scaleY(var(--transform-scale-y))",
  "@supports (display: grid)": {
    "&": {
      color: "#76a9fa",
    },
  },
  "&": {
    color: "#bcf0da !important",
    backgroundColor: "#f8b4b4 !important",
  },
  "@media (min-width: 1280px)": {
    "&:hover": {
      backgroundColor: "#e1effe",
      color: "#fbd5d5",
    },
  },
};


`;

exports[`macros no usage: no usage 1`] = `

import tw from '../lib/macro'

      ↓ ↓ ↓ ↓ ↓ ↓



`;

exports[`macros tailwind-ui ussage: tailwind-ui ussage 1`] = `

import tw from '../lib/macro';
const css = tw\`relative w-full flex justify-center py-2 px-4 border border-transparent text-sm leading-5 font-medium rounded-md text-white bg-indigo-600 hover:bg-indigo-500 focus:outline-none focus:border-indigo-700 focus:shadow-outline-indigo active:bg-indigo-300 transition duration-150 ease-in-out\`;

      ↓ ↓ ↓ ↓ ↓ ↓

const css = {
  position: "relative",
  width: "100%",
  display: "flex",
  justifyContent: "center",
  paddingTop: "0.5rem",
  paddingBottom: "0.5rem",
  paddingLeft: "1rem",
  paddingRight: "1rem",
  borderWidth: "1px",
  borderColor: "transparent",
  fontSize: "0.875rem",
  lineHeight: "1.25rem",
  fontWeight: "500",
  borderRadius: "0.375rem",
  color: "#ffffff",
  backgroundColor: "#5850ec",
  transitionProperty:
    "background-color, border-color, color, fill, stroke, opacity, box-shadow, transform",
  transitionDuration: "150ms",
  transitionTimingFunction: "cubic-bezier(0.4, 0, 0.2, 1)",
  "&:hover": {
    backgroundColor: "#6875f5",
  },
  "&:focus": {
    outline: "0",
    borderColor: "#5145cd",
    boxShadow: "0 0 0 3px rgba(180, 198, 252, 0.45)",
  },
  "&:active": {
    backgroundColor: "#b4c6fc",
  },
};


`;

exports[`macros test screen: test screen 1`] = `

import tw from '../lib/macro';
const css = tw\`md2[text-red-200] bg-red-300\`;

      ↓ ↓ ↓ ↓ ↓ ↓

const css = {
  backgroundColor: "#f8b4b4",
  "@media (min-width: 668px) and (max-width: 767px), (min-width: 868px)": {
    color: "#fbd5d5",
  },
};


`;

exports[`macros variant class not allowed: variant class not allowed 1`] = `

import tw from '../lib/macro';
const css = tw\`md[container] form-input\`

      ↓ ↓ ↓ ↓ ↓ ↓

MacroError: /home/arthie/projects/tailwindcssinjs/packages/macro/test/macro.test.js: Error: Variant class "md" not allowed with class "container" of type "component"

`;
